### PR TITLE
Code Insights: [FE] Use Monaco query input for insight query field

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/21339
Note: this PR is based on https://github.com/sourcegraph/sourcegraph/pull/28101

## Context
This PR just adopts MonacoQueryInput for code insights data series field needs and reuses this component in the creation UI form. 

## Known issues 
- Query field doesn't provide good styles for a disabled state (it isn't a big problem cause we're going to allow editing this field on the edit page)
- Suggestions panel doesn't work if we have a multi-line query. It works only on the first line. (I think it's a problem with our suggestions system deep in code intel hook. We will solve this problem later in a separate issue)

### CODE REVIEW

File: client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/form-series-input/DataSeriesQueryField.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    const { value, className, onChange, onBlur = noop, disabled, autoFocus } = props
+    const { enhancedThemePreference } = useThemeState()
+
+    const monacoOptions = useMemo(() => ({ ...MONACO_OPTIONS, readOnly: disabled }), [disabled])
+
+    return (
+        <LazyMonacoQueryInput
+            queryState={{ query: value, changeSource: QueryChangeSource.userInput }}
+            isLightTheme={enhancedThemePreference === ThemePreference.Light}
+            isSourcegraphDotCom={false}
```
Comment (tjkandala) [inline client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/form-series-input/DataSeriesQueryField.tsx:43]: Will this never be `true`? I think it affects suggestions

**Review by tjkandala — APPROVED** [review]
LGTM, I'm happy to have Monaco options as a prop now (for VS Code)

File: client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/form-series-input/DataSeriesQueryField.tsx (hunk 1 above)
Comment (vovakulikov) [reply client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/form-series-input/DataSeriesQueryField.tsx:43]: Yeah. Good catch. I added this at really beginning of this PR and should update this, thanks for noticing this.

Comment (fkling) [reply client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/form-series-input/DataSeriesQueryField.tsx:43]: Yeah this is used to show the static `<organization>/<repo>` suggestions, which are not relevant for deployed instances. It might be worth refactoring this at some point. `isSourcegraphDotCom` is certainly another candidate to move into the global store :D

File: client/web/src/search/input/MonacoQueryInput.tsx [hunk 2]
```diff
@@ -255,7 +310,6 @@ export const MonacoQueryInput: React.FunctionComponent<MonacoQueryInputProps> =
                 editor.revealPosition(position)
             }
         }
-        editor.focus()
```
Comment (unclejustin) [inline client/web/src/search/input/MonacoQueryInput.tsx:258]: Qustion: Is this not being used somewhere?

**Review by unclejustin — APPROVED** [review]

**Comment by unclejustin** [comment]
@vovakulikov LGTM 🌮  Just one question, and you still have some outstanding comments to address.

File: client/web/src/search/input/MonacoQueryInput.tsx (hunk 2 above)
Comment (vovakulikov) [reply client/web/src/search/input/MonacoQueryInput.tsx:258]: Yeah. Good catch. I removed that and couldn't find any regressions. Since this has been added by @fkling, Felix could you check this removal? Do you remember what is the purpose of this focus here?

**Review by novoselrok — APPROVED** [review]
> Suggestions panel doesn't work if we have a multi-line query. It works only on the first line. (I think it's a problem with our suggestions system deep in code intel hook. We will solve this problem later in a separate issue)

FYI, known issue here: https://github.com/sourcegraph/sourcegraph/issues/23152

The problem is that our query scanner/parser is not line aware and everything is considered to be a single line. That's why it doesn't work beyond the first line.

File: client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/form-series-input/DataSeriesQueryField.tsx (hunk 1 above)
Comment (vovakulikov) [reply client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/form-series-input/DataSeriesQueryField.tsx:43]: Aha. Thanks, @fkling. In this case, I think we need this `isSourcegraphDotCom` prop to be equal to false because the code insights query shouldn't have repo/org information there. We have a separate field setting for it.

File: client/web/src/search/input/MonacoQueryInput.tsx (hunk 2 above)
Comment (fkling) [reply client/web/src/search/input/MonacoQueryInput.tsx:258]: Ah... ok. It looks like I added this because when updating the query via the search reference panel the input needs to focus so that users can immediately edit the placeholder that was added. I think this can just be moved to the end of the ` case QueryChangeSource.searchReference` block. No idea why I didn't put it there in the first place.

Comment (vovakulikov) [reply client/web/src/search/input/MonacoQueryInput.tsx:258]: Nice, Thanks Felix!

**Review by limitedmage — COMMENTED** [review]
Why use `MonacoQueryInput` instead of `SearchBox` which also adds the context selector and toggles? 

Also, very minor: Why was file `global.ts` added?

**Comment by vovakulikov** [comment]
@limitedmage good catch about global.ts (it's slipped from me during rebasing main into this branch) 

Good question about`MonacoQueryInput` 👍 , TLDR; at the moment we don't support context in code insights queries and we SearchBox contains a search bar and other UI elements that we don't need for now in the creation UI. So we picked a kind of low-level abstraction for Monaco editor here (similar to what we've seen in notebook search components)

But I think you're right about what we could adopt the SearchBox component and I think we will do it eventually we just need to create a simple API for the search box component. At the moment this component isn't ready for simple reusing in different UI contexts cause it requires a lot of additional and sometimes unnecessary data in its props. 

A small example of it

```tsx
<SearchBox
   authenticatedUser={null}
   searchContextsEnabled={false}
   settingsCascade={EMPTY_SETTINGS_CASCADE}
   showSearchContext={false}
   setPatternType={noop}
   setCaseSensitivity={noop}
   hasUserAddedRepositories={false}
   hasUserAddedExternalServices={false}
   showSearchContextManagement={false}
   defaultSearchContextSpec=''
   setSelectedSearchContextSpec={noop}
   fetchAutoDefinedSearchContexts={fetchAutoDefinedSearchContexts}
   fetchSearchContexts={fetchSearchContexts}
   getUserSearchContextNamespaces={getUserSearchContextNamespaces}
   telemetryService={NOOP_TELEMETRY_SERVICE}
   editorClassName={classNames(className, styles.field)}
   hasSearchButton={false}
   
   queryState={{ query: value, changeSource: QueryChangeSource.userInput }}
   isLightTheme={enhancedThemePreference === ThemePreference.Light}
   isSourcegraphDotCom={false}
   preventNewLine={false}
   onChange={({ query }) => onChange(query)}
   patternType={SearchPatternType.regexp}
   caseSensitive={false}
   globbing={true}
   height="auto"
   onSubmit={noop}
   editorOptions={monacoOptions}
   autoFocus={autoFocus}
   onBlur={onBlur}
/>
```

So most of the first props here are redundant for code insights and I think for search notebook blocks use case. 
Usually, I don't mind passing many setting props but it seems that it's becoming really hard to follow what is going on here in this component with that amount of props. 

My thoughts on this. I think it's ok to have a Monaco component "family" directory, kind of different builds of query field input for different use cases. As long as they are placed in one directory and well documented. 
We also could go further and provide **compound components** for building different query inputs right in consumer code 

```tsx
<QueryInput {...high-level common props}
   <QueryInput.ContextSelect />
   <QueryInput.Input {...query input specific props}/>
   <QueryInput.Contorols />
</QueryInput>
```

I think it might be better to take into account how complex that component is and how many corner cases we handle with fine-tuning props now.
